added PCI-DSS to OCP on GCP (#19)
finos/ansible-cfi-ocp: ocp/gcp/05_implement_ocp_compliance_operator/implement_ocp_compliance_operator.md

rhcos4-nerc-cip      57s
```

> 7. The next step is to set up the compliance scanning using the CIS and PCI-DSS profile, the default profile that is being used will run scans on a daily base and, where possible. auto-remiate any complaince issues.

Reviewer: ```suggestion
7. The next step is to set up the compliance scanning using the CIS and PCI-DSS profiles. The default profile that is being used will run scans on a daily base and, where possible, auto-remediate any compliance issues.
```

```

```console
NAME                  PHASE         RESULT
cis-compliance-ocp4   RUNNING       NOT-AVAILABLE
pci-compliance-ocp4   RUNNING       NOT-AVAILABLE
pci-compliance-ocp4   RUNNING       NOT-AVAILABLE
pci-compliance-ocp4   PENDING       NOT-AVAILABLE
pci-compliance-ocp4   LAUNCHING     NOT-AVAILABLE
pci-compliance-ocp4   RUNNING       NOT-AVAILABLE
cis-compliance-ocp4   RUNNING       NOT-AVAILABLE
cis-compliance-ocp4   RUNNING       NOT-AVAILABLE
cis-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
pci-compliance-ocp4   RUNNING       NOT-AVAILABLE
pci-compliance-ocp4   RUNNING       NOT-AVAILABLE
pci-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
cis-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
cis-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
cis-compliance-ocp4   DONE          NON-COMPLIANT
cis-compliance-ocp4   DONE          NON-COMPLIANT
pci-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
pci-compliance-ocp4   AGGREGATING   NOT-AVAILABLE
pci-compliance-ocp4   DONE          NON-COMPLIANT
pci-compliance-ocp4   DONE          NON-COMPLIANT
```

> 9. Once the scans are complete the following command can be used to check the complaince scan results, when setting up the compliance scan object (step 7), we set the complaince operator to automatical remidate compliance failures. 

Reviewer: ```suggestion
9. Once the scans are complete, the following command can be used to check the compliance scan results. When setting up the compliance scan object (step 7), we set the compliance operator to automatically remediate compliance failures. 
```

ocp4-cis-scc-limit-privilege-escalation                                   MANUAL   medium
ocp4-cis-scc-limit-privileged-containers                                  MANUAL   medium
ocp4-cis-scc-limit-process-id-namespace                                   MANUAL   medium
ocp4-cis-scc-limit-root-containers                                        MANUAL   medium
ocp4-cis-secrets-consider-external-storage                                MANUAL   medium
ocp4-cis-secrets-no-environment-variables                                 MANUAL   medium
ocp4-pci-dss-accounts-restrict-service-account-tokens                     MANUAL   medium
ocp4-pci-dss-accounts-unique-service-account                              MANUAL   medium
ocp4-pci-dss-api-server-admission-control-plugin-alwaysadmit              PASS     medium
ocp4-pci-dss-api-server-admission-control-plugin-alwayspullimages         PASS     high
ocp4-pci-dss-api-server-admission-control-plugin-namespacelifecycle       PASS     medium
ocp4-pci-dss-api-server-admission-control-plugin-noderestriction          PASS     medium
ocp4-pci-dss-api-server-admission-control-plugin-scc                      PASS     medium
```

> Initially a number of the compliance checks will FAIL, many of them will be remediated automatically by the compliance operator. This remediation process can take sometime (hours). These remediations are actioned by the OCP cluster operators, using the following command the progress of these cluster operators can be seen.

Reviewer: ```suggestion
Initially a number of the compliance checks will fail, and many of them will be remediated automatically by the compliance operator. This process may take hours to complete. These remediations are actioned by the OCP cluster operators, using the following command the progress of these cluster operators can be seen.
```

'''shell
oc get co
```
Below is an example of the output of this command.

> #todo - update with new output when update to 4.13 takes place

Reviewer: is this TODO intended to stay in the PR? up to you, just double checking.

Author: Yes. thanks for highlighting.

oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-cis-node-worker,compliance.openshift.io/check-status=FAIL
```

### The OCP PCI Benchmark

The OCP PCI policy consists of two policies

- *ocp4-pci-dss* for Cluster polies
- *ocp4-pci-dss-node* for Node polies

> As with the CIS policy the *oc get compliancecheckresults* command can be used to check for scan results, for example

Reviewer: ```suggestion
As with the CIS policy the `oc get compliancecheckresults` command can be used to check for scan results. For example
```

```bash

oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-pci-dss,compliance.openshift.io/check-status=MANUAL

oc get compliancecheckresults -l compliance.openshift.io/scan-name=ocp4-pci-dss-node-worker,compliance.openshift.io/check-status=PASS

```

> Once any remediation have be made the following commands can be used to trigger a compliance rescan. A [script](rescan.sh) to trigger a rescan of all scans has been provided. 

Reviewer: ```suggestion
Once any remediation has been made, the following commands can be used to trigger a compliance rescan. A [script](rescan.sh) to trigger a rescan of all scans has been provided. 
```
